octopoes/octopoes/api/api.py

```diff
@@ -57,47 +58,56 @@
 
 
 @app.exception_handler(RequestValidationError)
-def http_validation_exception_handler(request: Request, exc: RequestValidationError) -> JSONResponse:
-    logger.critical(exc)
-    return JSONResponse(
-        {
-            "value": str(exc),
-        },
-        status.HTTP_422_UNPROCESSABLE_ENTITY,
-    )
+def http_validation_exception_handler(_: Request, exc: RequestValidationError) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(exc))
 
 
 @app.exception_handler(RequestError)
-def http_exception_handler(request: Request, exc: RequestError) -> JSONResponse:
-    logger.critical(exc)
-    return JSONResponse(
-        {
-            "value": str(exc),
-        },
-        status.HTTP_502_BAD_GATEWAY,
-    )
+def http_exception_handler(_: Request, exc: RequestError) -> None:
+    logger.error(exc)
+    raise HTTPException(status.HTTP_502_BAD_GATEWAY, detail=str(exc))
+
+
+@app.exception_handler(InvalidField)
+def invalid_field(_: Request, exc: InvalidField) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_400_BAD_REQUEST, detail=str(exc))
+
+
+@app.exception_handler(InvalidPath)
+def invalid_path(_: Request, exc: InvalidPath) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_400_BAD_REQUEST, detail=str(exc))
+
+
+@app.exception_handler(ValueError)
+def value_error(_: Request, exc: ValueError) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_400_BAD_REQUEST, detail=str(exc))
+
+
+@app.exception_handler(TypeNotFound)
+def type_not_found(_: Request, exc: TypeNotFound) -> None:
+    raise HTTPException(status.HTTP_404_NOT_FOUND, detail="Type not found")
+
+
+@app.exception_handler(NodeNotFound)
+def node_not_found_exception_handler(_: Request, exc: NodeNotFound) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_404_NOT_FOUND, detail="Node not found")
 
 
 @app.exception_handler(ObjectNotFoundException)
-def not_found_exception_handler(request: Request, exc: ObjectNotFoundException) -> JSONResponse:
+def not_found_exception_handler(_: Request, exc: ObjectNotFoundException) -> None:
     logger.info(exc)
-    return JSONResponse(
-        {
-            "value": exc.value,
-        },
-        status.HTTP_404_NOT_FOUND,
-    )
+    raise HTTPException(status.HTTP_404_NOT_FOUND, detail=str(exc))
 
 
 @app.exception_handler(Exception)
-def uncaught_exception_handler(request: Request, exc: Exception) -> JSONResponse:
-    logger.critical(exc)
-    return JSONResponse(
-        {
-            "value": f"{exc.__class__.__name__}: {exc}",
-        },
-        status.HTTP_500_INTERNAL_SERVER_ERROR,
-    )
+def uncaught_exception_handler(_: Request, exc: Exception) -> None:
+    logger.error(exc)
+    raise HTTPException(status.HTTP_404_NOT_FOUND, detail=f"{exc.__class__.__name__}: {exc}")
```

Comment:
This should keep returning a 500 and not a 404.

Reply:
Oh whoops, good catch! My bad.